Kernel/Jail.h

```diff
@@ -37,12 +38,21 @@ class Jail : public RefCounted<Jail> {
     SpinlockProtected<size_t, LockRank::None>& attach_count() { return m_attach_count; }
 
 private:
-    Jail(NonnullOwnPtr<KString>, JailIndex);
+    Jail(NonnullOwnPtr<KString>, JailIndex, ScopedProcessList&);
 
     NonnullOwnPtr<KString> m_name;
     JailIndex const m_index;
 
-    IntrusiveListNode<Jail, NonnullLockRefPtr<Jail>> m_jail_list_node;
+    IntrusiveListNode<Jail, NonnullLockRefPtr<Jail>> m_list_node;
+
+public:
+    using List = IntrusiveListRelaxedConst<&Jail::m_list_node>;
+
+    RefPtr<ScopedProcessList> process_list();
+
+private:
+    RefPtr<ScopedProcessList> m_process_list;
```

Comment:
Why is this nullable even though the constructor receives a reference?

Reply:
Later on we might want to be able to create a jail without isolating jailed processes from viewing other processes, which is quite useful for GUI apps. For now I will fix this to use a `NonnullRefPtr` :)